singer/src/main/java/com/pinterest/singer/writer/KafkaWritingTaskResult.java

```diff
@@ -15,12 +15,19 @@
  */
 package com.pinterest.singer.writer;
 
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import java.util.ArrayList;
+import java.util.List;
+
 public class KafkaWritingTaskResult {
 
   public final boolean success;
   public final Exception exception;
   private int writtenBytesSize;
   private int kafkaBatchWriteLatencyInMillis;
+  private List<RecordMetadata> recordMetadataList = new ArrayList<>();
```

Comment:
This is likely wasted as it's reset during the result capture part.

Reply:
+1    is this assignment needed?

Reply:
Initializing this field is to prevent NPE.  If not initialized and when KafkaWritingTask failed, this field of the returned KafkaWritingTaskResult will be NULL. accessing it cause NPE.

Thus, initializing is needed.  I added comments in the update of the PR